src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -523,12 +527,21 @@ public async Task ExtractManifestAsync(string nupkgPath, string targetPath)
                     string msiExtractionPath = Path.Combine(extractionPath, "msi");
 
 
-                    _ = WindowsInstaller.SetInternalUI(InstallUILevel.None);
-                    var result = WindowsInstaller.InstallProduct(msiPath, $"TARGETDIR={msiExtractionPath} ACTION=ADMIN");
-
-                    if (result != Error.SUCCESS)
+                    lock (_msiAdminInstallLock)
                     {
-                        throw new GracefulException(String.Format(LocalizableStrings.FailedToExtractMsi, msiPath));
+                        string adminInstallLog = Path.Combine(Path.GetTempPath(), $"Microsoft.NET.Workload_{DateTime.Now:yyyyMMdd_HHmmss}_AdminInstall_{Path.GetFileNameWithoutExtension(msiPath)}.log");
```

Comment:
Can you call into GetMsiLogName? (it's in MsiInstallerBase) You may need to add a new overload, but that will ensure that we have consistent names which helps to identify all the logs associated with a specific command execution.